### PR TITLE
Migrated makeshift actions from talon to extension

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Also merged implementation with some extension side actions that only called a vscode command

## Checklist

- [ ] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)

### CODE REVIEW

**Comment by phillco** [comment]
Wow, that was fast :)

File: cursorless-talon/src/actions/actions_simple.py [hunk 1]
```diff
@@ -46,6 +53,7 @@
 # Don't wait for these actions to finish, usually because they hang on some kind of user interaction
 no_wait_actions = [
     "generateSnippet",
+    "rename",
```
Comment (AndreasArvidsson) [inline cursorless-talon/src/actions/actions_simple.py:56]: Since we are not waiting for the rename action any sleep extension side doesn't really matter. Maybe we should turn this into an dict with the value being the sleep time out that we run talon side for that no waiting actions?

File: src/actions/ActionsMakeshift.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+export class ShowReferences extends MakeshiftAction {
+  command = "references-view.find";
+  ensureSingleTarget = true;
+}
+
+export class Rename extends MakeshiftAction {
+  command = "editor.action.rename";
+  ensureSingleTarget = true;
+  postCommandSleepMs = 200;
```
Comment (AndreasArvidsson) [inline src/actions/ActionsMakeshift.ts:63]: This sleep doesn't matter since talon is not waiting for this command

**Comment by AndreasArvidsson** [comment]
> Wow, that was fast :)

Well it was a small update :)

File: cursorless-talon/src/actions/actions_simple.py (hunk 1 above)
Comment (pokey) [reply cursorless-talon/src/actions/actions_simple.py:56]: Wait so does this mean that in your PR, rename no longer has any sleep?

File: src/actions/ActionsMakeshift.ts (hunk 2 above)
Comment (pokey) [reply src/actions/ActionsMakeshift.ts:63]: How come it is here then?

File: src/actions/ActionsMakeshift.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import { Target } from "../typings/target.types";
```
Comment (pokey) [inline src/actions/ActionsMakeshift.ts:1]: Why not just call this `MakeshiftActions`?

File: src/actions/ActionsMakeshift.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import sleep from "../util/sleep";
+import CommandAction from "./CommandAction";
+
+abstract class MakeshiftAction extends CommandAction {
+  abstract command: string;
```
Comment (pokey) [inline src/actions/ActionsMakeshift.ts:6]: Wow TIL.  Didn't realise you could declare non-function properties abstract 😄

File: src/actions/ActionsMakeshift.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  async run(targets: [Target[]]) {
+    return super.run(targets, {
+      command: this.command,
+      commandArgs: this.commandArg ? [this.commandArg] : [],
+      ensureSingleTarget: this.ensureSingleTarget ?? false,
+      restoreSelection: this.restoreSelection ?? true,
+    });
+    if (this.postCommandSleepMs) {
+      await sleep(this.postCommandSleepMs);
+    }
```
Comment (pokey) [inline src/actions/ActionsMakeshift.ts:21]: This code never runs

**Comment by pokey** [comment]
Overall looks good!  Left a few minor comments; biggest issues for me are around sleeps

File: src/actions/ActionsMakeshift.ts (hunk 4 above)
Comment (AndreasArvidsson) [reply src/actions/ActionsMakeshift.ts:6]: Cool isn't it? :D

File: src/actions/ActionsMakeshift.ts (hunk 2 above)
Comment (AndreasArvidsson) [reply src/actions/ActionsMakeshift.ts:63]: Because I wanted to have a discussion about it. This sleep is necessary for mac os afaik. So we have to put it somewhere.

File: cursorless-talon/src/actions/actions_simple.py (hunk 1 above)
Comment (AndreasArvidsson) [reply cursorless-talon/src/actions/actions_simple.py:56]: Well it sleeps extension side but since the client is not waiting for it so in practice no.

Comment (pokey) [reply cursorless-talon/src/actions/actions_simple.py:56]: Maybe just have a separate dict mapping from action id to sleep time. Not sure we want to fold it into the no-wait dict

Comment (AndreasArvidsson) [reply cursorless-talon/src/actions/actions_simple.py:56]: Done

File: cursorless-talon/src/actions/actions.py [hunk 6]
```diff
@@ -40,22 +40,10 @@ def cursorless_command(action_id: str, target: dict):
...
-
-            if talon_options.post_command_sleep_ms:
-                actions.sleep(f"{talon_options.post_command_sleep_ms}ms")
-
-            return return_value
         elif action_id in no_wait_actions:
-            return actions.user.cursorless_single_target_command_no_wait(
-                action_id, target
-            )
+            actions.user.cursorless_single_target_command_no_wait(action_id, target)
```
Comment (pokey) [inline cursorless-talon/src/actions/actions.py:44]: How come you're dropping the return value?

File: cursorless-talon/src/actions/actions.py [hunk 7]
```diff
@@ -40,22 +40,10 @@ def cursorless_command(action_id: str, target: dict):
...
+            if action_id in no_wait_actions_post_sleep:
+                actions.sleep(no_wait_actions_post_sleep[action_id])
```
Comment (pokey) [inline cursorless-talon/src/actions/actions.py:46]: How come we only sleep for actions that we don't wait for?  Seems somewhat orthogonal

**Review by pokey — COMMENTED** [review]
Ok getting there; left a couple more minor comments

File: cursorless-talon/src/actions/actions.py (hunk 7 above)
Comment (AndreasArvidsson) [reply cursorless-talon/src/actions/actions.py:46]: Action we wait for we sleep extension side.

File: cursorless-talon/src/actions/actions.py (hunk 6 above)
Comment (AndreasArvidsson) [reply cursorless-talon/src/actions/actions.py:44]: Since we don't wait it cannot return a value. In fact the action it's calling does not have a return.

**Review by pokey — APPROVED** [review]
lgtm!
